apps/prairielearn/src/components/SideNav.html.ts

```diff
@@ -0,0 +1,332 @@
+import { html, type HtmlValue } from '@prairielearn/html';
+
+import type { Course } from '../lib/db-types.js';
+
+import type { NavPage, NavSubPage } from './Navbar.types.js';
+import { ProgressCircle } from './ProgressCircle.html.js';
+
+interface SideNavTabInfo {
+  /** For the side nav tab to be active, the current navPage must be in activePages. */
+  activePages: NavPage[];
+  /**
+   * For navPages in checkActiveSubPageForPages, the current subPage must also be in activeSubPages for
+   * the side nav tab to be active.
+   *
+   * For all other navPages, only the current navPage must be in activePages.
+   *
+   * If not specified, activeSubPages will be checked on all pages.
+   **/
+  checkActiveSubPageForPages?: NavPage[];
+  /** For the side nav tab to be active, the navSubPage must be in activeSubPages. */
+  activeSubPages: NavSubPage[];
+  urlSuffix: string | ((resLocals: Record<string, any>) => string);
+  iconClasses: string;
+  tabLabel: string;
+  htmlSuffix?: (resLocals: Record<string, any>) => HtmlValue;
+  renderCondition?: (resLocals: Record<string, any>) => boolean;
+}
+
+const sideNavPagesTabs: Partial<Record<Exclude<NavPage, undefined>, SideNavTabInfo[]>> = {
+  instance_admin: [
+    {
+      activePages: ['instance_admin', 'assessments', 'assessment', 'assessment_instance'],
+      checkActiveSubPageForPages: ['instance_admin'],
+      activeSubPages: ['assessments'],
+      urlSuffix: '/instance_admin/assessments',
+      iconClasses: 'fa fa-list fa-fw',
+      tabLabel: 'Assessments',
+    },
+    {
+      activePages: ['instance_admin'],
+      activeSubPages: ['gradebook'],
+      urlSuffix: '/instance_admin/gradebook',
+      iconClasses: 'fas fa-balance-scale fa-fw',
+      tabLabel: 'Gradebook',
+      renderCondition: ({ authz_data }) => authz_data.has_course_instance_permission_view,
+    },
+    {
+      activePages: ['instance_admin'],
+      activeSubPages: ['file_view', 'file_edit'],
+      urlSuffix: '/instance_admin/file_view',
+      iconClasses: 'fa fa-edit fa-fw',
+      tabLabel: 'Files',
+    },
+    {
+      activePages: ['instance_admin'],
+      activeSubPages: ['lti13'],
+      urlSuffix: '/instance_admin/lti13_instance',
+      iconClasses: 'fas fa-school-flag fa-fw',
+      tabLabel: 'LTI 1.3',
+      renderCondition: (resLocals) => resLocals.lti13_enabled,
+    },
+    {
+      activePages: ['instance_admin'],
+      activeSubPages: ['settings', 'access', 'lti', 'billing'],
+      urlSuffix: '/instance_admin/settings',
+      iconClasses: 'fas fa-cog fa-fw',
+      tabLabel: 'Settings',
+    },
+  ],
+  course_admin: [
```

Comment:
Tiny nit: I'd be inclined to put this before `instance_admin` to reflect the fact that it comes higher in the navbar.